pkg/process/build/build_test.go

```diff
@@ -64,11 +78,151 @@ func TestBuild(t *testing.T) {
 		tags := []string{"1.3.1"}
 		semanticVersion := ""
 		completeVersion := ""
-		err := checkTagsForOverwrite(isSnapshot, tags, tagWith, semanticVersion, completeVersion)
+		err := process.CheckTagsForOverwrite(isSnapshot, tags, tagWith, semanticVersion, completeVersion)
 
 		if err != nil {
 			t.Fatal("Overwrite should be allowed for tagWith-snapshot")
 		}
 	})
 
+	t.Run("Test that build sends correct data to Sporingslogger ", func(t *testing.T) {
+
+		//mockgen -destination=pkg/docker/mocks/mock_registry.go -package=mocks -source=pkg/docker/registry.go Registry
+		//mockgen -destination=pkg/nexus/mocks/mock_nexus.go -package=mocks -source=pkg/nexus/nexus.go Downloader
+		// mockgen -destination=pkg/process/build/mocks/mock_layer_builder.go -package=mocks -source=pkg/process/build/build.go  Builder
+		//mockgen -destination=pkg/trace/mocks/mock_trace.go -package=mocks -source=pkg/trace/trace.go  Trace
+
+		ctx := context.Background()
+
+		appSpec := config.ApplicationSpec{
+			config.MavenGav{
+				"ArtifactId",
+				"GroupId",
+				"Version111",
+				"Classifier",
+				"Type",
+			},
+			config.DockerBaseImageSpec{
+				"BaseImage",
+				"BaseVersion",
+			},
+		}
+
+		dockerSpec := config.DockerSpec{
+			"OutputRegistry",
+			"OutputRepository",
+			"InternalPullRegistry",
+			config.PushExtraTags{
+				false,
+				true,
+				false,
+				false,
+			},
+			//This is the external docker registry where we check versions.
+			"ExternalDockerRegistry",
+			//The tag to push to. This is only used for ImageStreamTags (as for now) and RETAG functionality
+			"TagWith",
+			"RetagWith",
+		}
+
+		testConfig := config.Config{
+			"ApplicationType",
+			appSpec,
+			dockerSpec,
+			config.BuilderSpec{
+				"BaseImageVersion123",
+			},
+			true,
+			true,
+			true,
+			10,
+			false,
+			"Sporingstjeneste",
+			"OwnerReferenceUid",
+			"BinaryBuildType",
+			"NexusIQReportUrl",
+		}
+
+		mockCtrl := gomock.NewController(t)
+		registryClient := docker_mock.NewMockRegistry(mockCtrl)
+		nexusDownloader := nexus_mock.NewMockDownloader(mockCtrl)
+		layerBuilder := build_mock.NewMockBuilder(mockCtrl)
+		traceMock := trace_mock.NewMockTrace(mockCtrl)
+
+		registryClient.EXPECT().GetImageInfo(gomock.Any(), gomock.Any(), gomock.Any()).Return(&runtime.ImageInfo{
```

Comment:
Mye Any() i mockene her..

Reply:
Laget test for å se at riktig data blir sendt til sporingslogger ved bygg. 
Så hva som blir sendt inn til registry etc er ikke viktig for akkurat dette.